src/sinks/pulsar.rs

```diff
@@ -46,11 +47,15 @@ pub enum Encoding {
     Json,
 }
 
+enum PulsarProducerState {
+    NotReady(PulsarSinkConfig),
+    Creating(Box<dyn Future<Item = Producer<TokioExecutor>, Error = ()> + Send>),
+    Ready(Arc<Mutex<Producer<TokioExecutor>>>),
```

Comment:
Is the `Arc<Mutex<_>>` needed? It looks like `Producer` is `Send` and `Sync`: https://docs.rs/pulsar/1.0.0-alpha-4/pulsar/producer/struct.Producer.html#synthetic-implementations

Reply:
:thinking: we need `&mut self` for [send method](https://docs.rs/pulsar/1.0.0-alpha-4/pulsar/producer/struct.Producer.html#method.send). I'm not sure, can we get it without `Arc<Mutex<_>>`?